src/niamoto/core/components/importers/shapes.py

```
@@ -50,116 +50,229 @@ def import_from_config(self, shapes_config: List[Dict[str, Any]]) -> str:
+                                                    import_stats["updated"] += 1
+
+                                        except Exception as feat_error:
+                                            import_stats["errors"].append(
+                                                f"Error processing feature: {str(feat_error)}"
+                                            )
+                                            import_stats["skipped"] += 1
+
+                            except fiona.errors.DriverError as e:
+                                raise ValueError(
+                                    f"Unable to open file {actual_file_path}: {str(e)}"
+                                )
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)